pulsar-transaction/buffer/src/test/java/org/apache/pulsar/transaction/buffer/impl/PersistentTransactionBufferTest.java

```diff
@@ -0,0 +1,726 @@
+
+        // call openCursorComplete on cursor asyncOpen
+        doAnswer(new Answer<Object>() {
+            @Override
+            public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
+                ((OpenCursorCallback) invocationOnMock.getArguments()[2]).openCursorComplete(cursorMock, null);
+                return null;
+            }
+        }).when(ledgerMock)
+          .asyncOpenCursor(matches(".*success.*"), any(InitialPosition.class), any(OpenCursorCallback.class), anyObject());
+
+        doAnswer(new Answer<Object>() {
+            @Override
+            public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
+                ((OpenCursorCallback) invocationOnMock.getArguments()[3]).openCursorComplete(cursorMock, null);
+                return null;
+            }
+        }).when(ledgerMock).asyncOpenCursor(matches(".*success.*"), any(InitialPosition.class), any(Map.class),
+                                            any(OpenCursorCallback.class), anyObject());
+
+        // call deleteLedgerComplete on ledger asyncDelete
+        doAnswer(new Answer<Object>() {
+            @Override
+            public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
+                ((DeleteLedgerCallback) invocationOnMock.getArguments()[0]).deleteLedgerComplete(null);
+                return null;
+            }
+        }).when(ledgerMock).asyncDelete(any(DeleteLedgerCallback.class), anyObject());
+
+        doAnswer(new Answer<Object>() {
+            @Override
+            public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
+                ((DeleteCursorCallback) invocationOnMock.getArguments()[1]).deleteCursorComplete(null);
+                return null;
+            }
+        }).when(ledgerMock).asyncDeleteCursor(matches(".*success.*"), any(DeleteCursorCallback.class), anyObject());
+
+        doAnswer((invokactionOnMock) -> {
+            ((MarkDeleteCallback) invokactionOnMock.getArguments()[2])
+                .markDeleteComplete(invokactionOnMock.getArguments()[3]);
+            return null;
+        }).when(cursorMock).asyncMarkDelete(anyObject(), anyObject(), any(MarkDeleteCallback.class), anyObject());
+
+        this.buffer = new PersistentTransactionBuffer(successTopicName, factory.open("hello"), brokerService);
+    }
+
+        @AfterMethod
+    public void teardown() throws Exception {
+        brokerService.getTopics().clear();
+        brokerService.close(); //to clear pulsarStats
+        try {
+            pulsar.close();
+        } catch (Exception e) {
+            log.warn("Failed to close pulsar service", e);
+            throw e;
+        }
+    }
+
+    private final TxnID txnID = new TxnID(1234L, 5678L);
+    private PersistentTransactionBuffer buffer;
+
+    @Test
+    public void testGetANonExistTxn() {
+        buffer.getTransactionMeta(txnID).whenComplete(((meta, throwable) -> {
```

Comment:
this operation is asynchronous. you can't really test this method.

change it to

```
try {
     buffer.getTransactionMeta(txnID).get();
} catch (ExecutionException ee) {
     assertTrue(ee.getCause() instance TransactionNotFoundException);
}
```

Reply:
Also please fix all the other tests as well.